dotnet-desktop-guide/net/winforms/wfdev-diagnostics/wfdev006.md

```diff
@@ -0,0 +1,65 @@
+---
+title: WFDEV006 warning
+description: Learn about Windows Forms compile-time warning WFDEV006. ContextMenu, DataGrid, MainMenu, Menu, StatusBar, ToolBar are obsolete. They're provided for binary compatibility with .NET Framework.
+ms.date: 03/04/2025
+f1_keywords:
+  - "WFDEV006"
+helpviewer_keywords:
+  - "WFDEV006"
+---
+# Compiler Warning WFDEV006
+
+**Version introduced:** .NET 10 Preview 1
+
+> `ContextMenu`, `DataGrid`, `MainMenu`, `Menu`, `StatusBar`, `ToolBar` are obsolete. They're provided for binary compatibility with .NET Framework.
+
+Starting with .NET 10, some controls are provided for binary compatibility with .NET Framework, but they're marked as obsolete and aren't intended to be used directly from your code. Referencing one of the following controls generates warning `WFDEV006` at compile time:
+
+- <xref:System.Windows.Forms.ContextMenu>
+- <xref:System.Windows.Forms.DataGrid>
+- <xref:System.Windows.Forms.MainMenu>
+- <xref:System.Windows.Forms.Menu>
+- <xref:System.Windows.Forms.StatusBar>
+- <xref:System.Windows.Forms.ToolBar>
+
+Starting with .NET 10, these controls are provided for binary compatibility with .NET Framework, but they're marked as obsolete and aren't intended to be used directly from your code. Use a replacement control instead. Referencing one of these controls warning `WFDEV006` at compile time.
+
+## Workaround
+
+Replace references to these controls with their .NET counterparts:
+
+| Original control | New control                                    |
+|------------------|------------------------------------------------|
+| `ContextMenu`    | `ContextMenuStrip`                             |
+| `DataGrid`       | `DataGridView`                                 |
+| `MainMenu`       | `MenuStrip`                                    |
+| `Menu`           | `ToolStripDropDown` or `ToolStripDropDownMenu` |
+| `StatusBar`      | `StatusStrip`                                  |
+| `ToolBar`        | `ToolStrip`                                    |
+
+## Suppress a warning
+
+If you must use the obsolete APIs, you can suppress the warning in code or in your project file.
```

Comment:
```suggestion
If you must reference the obsolete APIs, for example to run reflection on them, you can suppress the warning in code or in your project file. Note that these types can't be instantiated. They are provided to support .NET applications that reference .NET Framework 3rd party libraries that use these types and can't be updated. The new code will JIT correctly and if the unsupported code is not executed, the application will run instead of JIT throwing a missing member exception.
```

Reply:
I want to say that there is hardly a reason to suppress this warning.

Reply:
Good idea.

Reply:
Whoops. Looks like a previous update I hadn't yet pushed erased this. I'll manually add it.